calculate_genweights.py

```diff
@@ -44,3 +47,40 @@ def calculate_genweight(dataset):
     except:
         print("Error when reading input files")
         return 1.0
+
+
+def calculate_genweight_uproot(dataset):
+    print(f"Counting negative and positive genweights for {dataset['nick']}...")
+    filelist = read_filelist_from_das(dataset["dbs"])
+    negative = 0
+    positive = 0
+    # set a threshold that if more than 10% of the files fail, the function returns None
+    threshold = len(filelist) // 10
+    fails = 0
+
+    print(f"Threshold for failed files: {threshold}")
+    print(f"Number of files: {len(filelist)}")
+    # loop over all files and count the number of negative and positive genweights
+    with Progress() as progress:
+        task = progress.add_task("Files read ", total=len(filelist))
+        filelist = [file + ":Events" for file in filelist]
+        for i, file in enumerate(filelist):
+            try:
+                events = uproot.open(file)
```

Comment:
We do not know why the reading of the files fails, i.e. whether it is a problem of the site that persists for some time or just a temporary issue when reading the file, right? In the latter case we could possibly try to add a second or a few more tries to read the file.

Reply:
I can add a timeout of ~5 sec for the open operation, what do you think about that ?

Reply:
Maybe this helps. I could also check why it fails using the DEBUG output first. Do you know how long the current timeout is?